packages/extension/src/common/keyNavigationService.js

```diff
@@ -17,75 +17,56 @@ export const validKeys = {
   esc: 27,
 };
 
-function getNewPostEl(keyCode, currentElement, insaneMode) {
-  if (keyCode === validKeys.h && !insaneMode) return getAbovePost(currentElement);
+const validKeysValues = Object.values(validKeys);
 
-  if (keyCode === validKeys.l && !insaneMode) return getBelowPost(currentElement);
+function navigateFeed(keyCode, currentElement, insaneMode) {
+  if (keyCode === validKeys.h && !insaneMode) return getPreviousPost(currentElement);
 
-  if (keyCode === validKeys.j && insaneMode) return getBelowPost(currentElement);
+  if (keyCode === validKeys.l && !insaneMode) return getNextPost(currentElement);
 
-  if (keyCode === validKeys.k && insaneMode) return getAbovePost(currentElement);
+  if (keyCode === validKeys.j) return getBelowPost(currentElement, insaneMode);
+
+  if (keyCode === validKeys.k) return getAbovePost(currentElement, insaneMode);
 
   return currentElement;
 }
 
-function triggerBookmark(post) {
-  if (post.ad) return null;
+function triggerBookmark(article) {
+  if (article.ad) return null;
 
-  return post.$emit('bookmark', { post: post.post, bookmarked: !post.post.bookmarked });
+  return article.$emit('bookmark', { post: article.post, bookmarked: !article.post.bookmarked });
 }
 
-function getCurrentPost(posts, current) {
-  if (!current) return null;
-
-  const postOrAdProp = current.post || current.ad;
+function navigateDaily({ keyCode }) {
+  if (validKeysValues.indexOf(keyCode) === -1) return null;
 
-  return posts.find(article => [article.ad, article.post].indexOf(postOrAdProp) !== -1);
-}
-
-function navigate(keyCode, posts, search, backToMainFeed, { current, insaneMode }) {
-  if (keyCode === validKeys.esc) return backToMainFeed();
+  const { daFeedRef, hoveredPost } = store.state.feed;
```

Comment:
I don't like saving references in the store. Maybe it's better to initialize the key navigation service with callback functions and once something happens instead of using the refs, the service will call these callbacks.
I find current implementation too coupled to the components and their functions. By using callbacks it will be agnostic.

Reply:
I felt too that it is having dependencies on so many places especially the reference of a whole component. Will definitely work on your suggestion in making it agnostic.